src/infra/ChatAnalyticsReporter.ts

```diff
@@ -0,0 +1,74 @@
+import type { EventPayload } from '../models/chat';
+import type ChatAnalyticsConfig from '../models/config/ChatAnalyticsConfig';
+
+interface EventsAPIResponse {
+  id: string;
+}
+
+/**
+ * A class to report chat analytics data. Uses the provided API key, environment,
+ * and partition to determine the endpoint for reporting events.
+ */
+export default class ChatAnalyticsReporter {
+  public readonly apiKey: string;
+  public readonly env: NonNullable<ChatAnalyticsConfig['env']>;
+  public readonly partition: NonNullable<ChatAnalyticsConfig['partition']>;
+  public readonly endpoint: string;
+
+  private endpoints: Record<
+    NonNullable<ChatAnalyticsConfig['partition']>,
+    Record<NonNullable<ChatAnalyticsConfig['env']>, string>
+  > = {
+    US: {
+      PROD: 'https://www.us.yextevents.com/accounts/me/events',
+      SANDBOX: 'https://www.sbx.us.yextevents.com/accounts/me/events',
+      QA: 'https://www.qa.us.yextevents.com/accounts/me/events',
+    },
+    EU: {
+      PROD: 'https://www.eu.yextevents.com/accounts/me/events',
+      SANDBOX: 'https://www.sbx.eu.yextevents.com/accounts/me/events',
+      QA: 'https://www.qa.eu.yextevents.com/accounts/me/events',
+    }
+  };
+
+  constructor({ apiKey, env, partition }: ChatAnalyticsConfig) {
+    this.apiKey = apiKey;
+    this.env = env ?? 'PROD';
+    this.partition = partition ?? 'US';
+    this.endpoint = this.endpoints[this.partition][this.env];
+  }
+
+  /**
+   * Report an event to the chat analytics API.
+   * @param event The event to report.
+   * @returns A promise that resolves to the response from the API.
+   * @example
+   * reporter.report({
+   *  action: 'DIRECTIONS',
+   *   chat: {
+   *   botId: 'davish-playground',
+   * },
+   * });
```

Comment:
Is there currently some provision to specify certain properties as always-passed by default?

It can wait for the final generic provideAnalytics API I guess. I was thinking about something which allows a form of currying, so the ChatAnalyticsReporter can be wrapped in reporter objects which add layers of defaults as code becomes more specific to a single site, a dingle experience, a single bot, etc.

Reply:
Yeah this is BY FAR the most frustrating part of working with this SDK. One approach would be to add a `defaultPayload` on initialization, and then we would deep merge that payload with any payload provided directly to `.report()`.

What does everyone think of that? @AjayBenno

Reply:
I think we should address this in a later follow up so we can get this out the door.